distsql/select_result.go

```diff
@@ -41,7 +41,7 @@ var (
 // SelectResult is an iterator of coprocessor partial results.
 type SelectResult interface {
 	// Fetch fetches partial results from client.
-	Fetch(context.Context)
+	Fetch(context.Context, tipb.EncodeType)
```

Comment:
We can put EncodeType in SessionVars and fetch it when building the selectResult.